### PR TITLE
DP-38: CodeBuild appspec.yml

### PR DESCRIPTION
**Pull request by acarlton**
Adds a CodeBuild task to the pipeline which checks for the existence of an appspec.yml and writes out a default one, along with BeforeInstall and AfterInstall scripts to change Drupal permissions, manage the symlink to EFS and clear the Drupal cache via Drush if it is installed.

### CODE REVIEW

File: cdk/drupal/drupal_stack.py [hunk 1]
```diff
@@ -1284,6 +1285,61 @@ def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
...
+                            resources=['*']
+                        ),
+                        aws_iam.PolicyStatement(
+                            effect=aws_iam.Effect.ALLOW,
+                            actions=[
+                                's3:GetObject',
+                                's3:PutObject'
+                            ],
+                            resources=[
+                                "arn:aws:s3:::github-user-and-bucket-taskcatbucket-2zppaw3wi3sx/*"
```
Comment (dylanvaughn) [inline cdk/drupal/drupal_stack.py:1312]: should be a parameter.

File: cdk/drupal/drupal_stack.py [hunk 2]
```diff
@@ -1284,6 +1285,61 @@ def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
...
+        with open('drupal/scripts/codebuild_transform_project_buildspec.yml') as f:
+            codebuild_transform_project_buildspec = f.read()
+        codebuild_transform_project = aws_codebuild.CfnProject(
+            self,
+            "CodeBuildTransformProject",
+            artifacts=aws_codebuild.CfnProject.ArtifactsProperty(
+                type="CODEPIPELINE",
+            ),
+            # TODO: add kms encryption key
+            # encryption_key="",
```
Comment (dylanvaughn) [inline cdk/drupal/drupal_stack.py:1328]: can we use the default AWS KMS key here?

Comment (acarlton) [reply cdk/drupal/drupal_stack.py:1328]: @dylanvaughn I'm not quite sure what you want here. From the console there are no customer-created KMS keys (our stack doesn't include one) and there a number of "default" keys specific to various aws services -- see screenshot. Let me know if we want to re-use a key for another service, or if we want to create one master key as part of the pending encryption ticket and reference it everywhere in the stack.

aws-kms-screenshot

File: cdk/drupal/drupal_stack.py (hunk 1 above)
Comment (acarlton) [reply cdk/drupal/drupal_stack.py:1312]: made this a conditional reference using Arn.format

File: cdk/drupal/drupal_stack.py (hunk 2 above)
Comment (dylanvaughn) [reply cdk/drupal/drupal_stack.py:1344]: Yeah let's address in a separate PR

**Pull request merged**